### PR TITLE
Handle repeating class names where order matters

### PR DESCRIPTION
**Pull request by talha-asad**
Fix for #30 
Submitted again with a slightly better comment and is no longer a complex one-liner. Has been tested locally and by a few other developers but no actual tests are included.

### CODE REVIEW

File: packages/blaze/attrs.js [hunk 1]
```diff
@@ -101,10 +101,18 @@ var ClassHandler = Blaze._DiffingAttributeHandler.extend({
...
-        tokens[token] = token;
+    var similarTokens = 0;
+    
+    _.each(attrString.split(' '), function(token, i) {
+      if (token) {
+        
+        // Incase of duplicate class names where order matters (e.g: Semantic UI), group similar tokens accordingly
+        if (tokens[token] && i > 0) {
+          var groupWithIndex = (i - similarTokens++) - 1;
+          tokens[Object.keys(tokens)[groupWithIndex]] += ' ' + token;
```
Comment (mitar) [inline packages/blaze/attrs.js]: Order of keys in an object is not defined in JavaScript.

**Comment by mitar** [comment]
So, why is this change really required and simply Meteor packages for Semantic UI do not [do the following change](https://github.com/Semantic-Org/Semantic-UI/issues/1526#issuecomment-151857979) can package the package in this way?

**Comment by talha-asad** [comment]
Semantic UI is not at fault here and it makes no sense to have special class syntax for Semantic UI in meteor because blaze needs patching.

Maybe order of keys in an object in javascript is not defined or the same in all cases, but that is a neat hack to fix the issue, which has been working nicely for me for the last few Blaze versions. Perhaps you can rewrite it in a way that will be consistent.

**Comment by mitar** [comment]
> Semantic UI is not at fault here 

From my reading of the HTML5 spec it is:

>  The classes that an HTML element has assigned to it consists of all the classes returned when the value of the class attribute is split on spaces. **(Duplicates are ignored.)**

(Emphasis mine.)

And Meteor is ignoring duplicates per spec.

**Comment by mitar** [comment]
> but that is a neat hack to fix the issue

Yes, sorry, we do not need neat hacks but working code. Readable and no-hacky code with tests is more important than a quick hacky fix.

**Comment by talha-asad** [comment]
After reading the spec, i agree with you. Lets patch Semantic UI ;)

**Comment by mitar** [comment]
So intermediary solution would be that Meteor packages fix this. It seems [people are confirming](https://github.com/Semantic-Org/Semantic-UI/issues/1526#issuecomment-217048865) that a change works. This could be part of the bundling process when packaging semantic UI for Meteor.

**Pull request closed without merging**